docs/src/FAQ.md

```diff
@@ -76,3 +76,88 @@ ChainRulesTestUtils.jl has some dependencies, so it is a separate package from C
 This means your package can depend on the light-weight ChainRulesCore.jl, and make ChainRulesTestUtils.jl a test-only dependency.
 
 Remember to read the section on [On writing good `rrule` / `frule` methods](@ref).
+
+## How do chain rules work for complex functions?
+
+`ChainRules.jl` follows the convention that `frule` applied to a function ``f(x + i y) = u(x,y) + i \, v(x,y)`` with perturbation ``\Delta x + i \Delta y`` returns the value and
+```math
+\tfrac{\partial u}{\partial x} \, \Delta x + \tfrac{\partial u}{\partial y} \, \Delta y + i \, \Bigl( \tfrac{\partial v}{\partial x} \, \Delta x + \tfrac{\partial v}{\partial y} \, \Delta y \Bigr)
+.
+```
+Similarly, `rrule` applied to the same function evaluates the function and returns a pullback which, when applied to the adjoint ``\Delta u + i \Delta v``, returns
+```math
+\Delta u \, \tfrac{\partial u}{\partial x} + \Delta v \, \tfrac{\partial v}{\partial x} + i \, \Bigl(\Delta u \, \tfrac{\partial u }{\partial y} + \Delta v \, \tfrac{\partial v}{\partial y} \Bigr)
+.
+```
+If we interpret complex numbers as vectors in ``\mathbb{R}^2``, then these rules correspond to left- and right-multiplication with the Jacobian of ``f(z)``, i.e. `frule` corresponds to
+```math
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} \, \Delta x + \tfrac{\partial u}{\partial y} \, \Delta y
+\\
+\tfrac{\partial v}{\partial x} \, \Delta x + \tfrac{\partial v}{\partial y} \, \Delta y
+\end{pmatrix}
+=
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} & \tfrac{\partial u}{\partial y} \\
+\tfrac{\partial v}{\partial x} & \tfrac{\partial v}{\partial y} \\
+\end{pmatrix}
+\begin{pmatrix}
+\Delta x \\ \Delta y
+\end{pmatrix}
+
+```
+and `rrule` corresponds to
```

Comment:
I assume you mean the (LaTeX) matvec formula for `rrule` here. I like the current right-multiplication interpretation of `rrule` since that's how I think about reverse differentiation, but you have a point as well. Someone just needs to make a call on this.

Reply:
Hmm good question. I think probably the right thing to do is to look at what we compute in practice. If I'm not mistaken, we compute `J^T v` so vote that. That way around would also be more consistent with the multi-dimensional case below.

Reply:
I agree. Though the vjp notation is more consistent with how we usually write the math, it's transposed from the code and risks introducing bugs.

Reply:
"Follow the code structure" (rather than "follow the maths structure") makes sense to me. Updated this.